Source/FluentMockContext.cs

```diff
@@ -90,6 +90,7 @@ public MockInvocation LastInvocation
 
 		public void Dispose()
 		{
+			invocations.Reverse();
```

Comment:
Out of curiosity (I haven't yet looked at this in detail), can you briefly explain why this works?